compiler-core/src/language_server/tests/action.rs

```diff
@@ -1560,6 +1561,435 @@ pub fn main(res) {
+    wobble.Wubble(1) -> 2
+  }
+  let wob = wobble.Wibble
+  todo
+}
+"#;
+    assert_code_action!(
+        CONVERT_TO_UNQUALIFIED_IMPORT,
+        TestProject::for_source(src)
+            .add_hex_module("wobble", "pub type Wobble { Wibble Wubble(Int) }"),
+        find_position_of("wobble.W").select_until(find_position_of("ibble"))
+    );
+}
+
+#[test]
+fn test_qualified_to_unqualified_import_type() {
+    let src = r#"
+import option
+
+pub fn main(x) -> option.Option(Int) {
+    option.Some(1)
+}
+"#;
+    assert_code_action!(
+        CONVERT_TO_UNQUALIFIED_IMPORT,
+        TestProject::for_source(src)
+            .add_hex_module("option", "pub type Option(v) { Some(v) None }"),
+        find_position_of(".Option").select_until(find_position_of("(Int)")),
+    );
+}
+
+#[test]
+fn test_qualified_to_unqualified_import_nested_type_outer() {
+    let src = r#"
+import option
+import wobble
+pub fn main(x) -> option.Option(wobble.Wibble) {
+    todo
+}
+"#;
+    assert_code_action!(
+        CONVERT_TO_UNQUALIFIED_IMPORT,
+        TestProject::for_source(src)
+            .add_hex_module("option", "pub type Option(v) { Some(v) None }")
+            .add_hex_module("wobble", "pub type Wibble { Wobble(Int) }"),
+        find_position_of(".O").select_until(find_position_of("ption(")),
+    );
+}
+
+#[test]
+fn test_qualified_to_unqualified_import_nested_constructor_outer() {
+    let src = r#"
+import option
+import wobble
+pub fn main(x) -> option.Option(wobble.Wibble) {
+    option.Some(wobble.Wobble(1))
+}
+"#;
+    assert_code_action!(
+        CONVERT_TO_UNQUALIFIED_IMPORT,
+        TestProject::for_source(src)
+            .add_hex_module("option", "pub type Option(v) { Some(v) None }")
+            .add_hex_module("wobble", "pub type Wibble { Wobble(Int) }"),
+        find_position_of(".S").select_until(find_position_of("ome(")),
+    );
+}
+
+#[test]
+fn test_qualified_to_unqualified_import_nested_constructor_inner() {
+    let src = r#"
+import option
+import wobble
+
+pub fn main(x) -> option.Option(wobble.Wibble) {
+    option.Some(wobble.Wobble(1))
+}
+"#;
+    assert_code_action!(
+        CONVERT_TO_UNQUALIFIED_IMPORT,
+        TestProject::for_source(src)
+            .add_hex_module("option", "pub type Option(v) { Some(v) None }")
+            .add_hex_module("wobble", "pub type Wibble { Wobble(Int) }"),
+        find_position_of(".Wobble").select_until(find_position_of("(1)")),
+    );
+}
+
+#[test]
+fn test_qualified_to_unqualified_import_nested_type_inner() {
+    let src = r#"
+import option
+import wobble
+
+pub fn main(x) -> option.Option(wobble.Wibble) {
+    todo
+}
+"#;
+    assert_code_action!(
+        CONVERT_TO_UNQUALIFIED_IMPORT,
+        TestProject::for_source(src)
+            .add_hex_module("option", "pub type Option(v) { Some(v) None }")
+            .add_hex_module("wobble", "pub type Wibble { Wobble(Int) }"),
+        find_position_of("wobble.").select_until(find_position_of("Wibble")),
+    );
+}
+
```

Comment:
Could you add tests for other module names, for other constructor names, for the import being over multiple lines, and for the import to come after the usages. Thank you

Reply:
More tests are added including other all above and bad formated import, usage inside function call, list or tuple and within complex patterns